elliptic-curve/src/ops.rs

```diff
@@ -1,78 +1,77 @@
 //! Traits for arithmetic operations on elliptic curve field elements.
 
+use core::iter;
 pub use core::ops::{Add, AddAssign, Mul, Neg, Shr, ShrAssign, Sub, SubAssign};
 pub use crypto_bigint::Invert;
 
 use crypto_bigint::Integer;
-use subtle::{Choice, ConditionallySelectable, CtOption};
+use ff::Field;
+use subtle::{Choice, CtOption};
 
 #[cfg(feature = "alloc")]
-use alloc::vec::Vec;
+use alloc::{borrow::ToOwned, vec::Vec};
 
 /// Perform a batched inversion on a sequence of field elements (i.e. base field elements or scalars)
 /// at an amortized cost that should be practically as efficient as a single inversion.
-pub trait BatchInvert<FieldElements: ?Sized>: Invert + Sized {
+pub trait BatchInvert<FieldElements: ?Sized>: Field + Sized {
     /// The output of batch inversion. A container of field elements.
     type Output: AsRef<[Self]>;
 
     /// Invert a batch of field elements.
     fn batch_invert(
         field_elements: &FieldElements,
     ) -> CtOption<<Self as BatchInvert<FieldElements>>::Output>;
```

Comment:
It seems like you could just have a default impl here:

```suggestion
    ) -> CtOption<<Self as BatchInvert<FieldElements>>::Output> {
        let mut field_elements_inverses = *field_elements;
        let inversion_succeeded = Self::batch_invert_mut(&mut field_elements_inverses);
        CtOption::new(field_elements_inverses, inversion_succeeded)
    }
```

Six of one, half dozen of another though

Reply:
Hm, but it would require `ToOwned`, which is behind `alloc` unfortunately. Not sure how else to pull this of.

We could change `batch_invert` to take ownership instead of a reference and remove `batch_invert_mut()` entirely?